src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -683,6 +685,23 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
+        internal async Task FlushAsync() // no cancellation token: we don't want individual operation tokens applying to the whole connection
```

Comment:
I wonder if we could implement this by just calling StartWriteAsync(0) and then FinishWrite(Now). Thoughts?

Reply:
Why is that better?  This implementation is a strict subset of what StartWriteAsync does.

Reply:
Ah, I see your _inProgressWrite above. Hmm.